Battleship/BattleshipGame.cs

```diff
@@ -152,10 +178,25 @@ protected override void Draw(GameTime gameTime)
         private void HandleShooting()
         {
             MouseState mouseState = Mouse.GetState();
-            if (mouseState.LeftButton == ButtonState.Pressed)
+            if (_shipManager!.ReadClick == true && mouseState.LeftButton == ButtonState.Pressed)
             {
+                _shipManager.ReadClick = false;
+                bool? success = false;
                 Point mousePoint = new Point(mouseState.X, mouseState.Y);
-                bool? hit = _player2grid!.Shoot(mousePoint);
+                if (_turnManager!.IsP1sTurn)
+                {
+                    success = _player2grid!.Shoot(mousePoint);
+                }
+                else if (!_turnManager.IsP1sTurn)
```

Comment:
```suggestion
                else
```
that behavior is implicit if the first if check fails.